streaming/store.go

```diff
@@ -0,0 +1,248 @@
+	client, err := utils.GetClient(ctx, dbURI)
+	if err != nil {
+		err = fmt.Errorf("can't create client for db %s: %v", dbURI, err)
+		return err
+	}
+	defer client.Close()
+	if err != nil {
+		err = fmt.Errorf("can't create database client: %v", err)
+		return err
+	}
+	err = writeJobDetails(ctx, migrationJobId, isSharded, conv, dbName, time.Now(), client)
+	if err != nil {
+		err = fmt.Errorf("can't store generated resources for datashard: %v", err)
+		return err
+	}
+	logger.Log.Info(fmt.Sprintf("Generated resources stored successfully for migration jobId: %s. You can also look at the 'spannermigrationtool_metadata' database in your spanner instance to get this jobId at a later point of time.\n", migrationJobId))
+	return nil
+}
+
+func PersistAggregateMonitoringResources(ctx context.Context, targetProfile profiles.TargetProfile, sourceProfile profiles.SourceProfile, conv *internal.Conv, migrationJobId string) error {
+	logger.Log.Debug(fmt.Sprintf("Storing aggregate monitoring dashboard for migration jobId: %s\n", migrationJobId))
+	project, instance, _, err := targetProfile.GetResourceIds(ctx, time.Now(), sourceProfile.Driver, nil)
+	if err != nil {
+		err = fmt.Errorf("can't get resource ids: %v", err)
+		return err
+	}
+	dbURI := fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, constants.METADATA_DB)
+	client, err := utils.GetClient(ctx, dbURI)
+	if err != nil {
+		err = fmt.Errorf("can't create client for db %s: %v", dbURI, err)
+		return err
+	}
+	defer client.Close()
+	aggMonitoringResourcesBytes, err := json.Marshal(conv.Audit.StreamingStats.AggMonitoringResources)
+	if err != nil {
+		logger.Log.Error(fmt.Sprintf("internal error occurred while persisting metadata for migration job %s: %v\n", migrationJobId, err))
+		return err
+	}
+	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		mutation, err := createResourceMutation(migrationJobId, constants.DEFAULT_SHARD_ID, conv.Audit.StreamingStats.AggMonitoringResources.DashboardName, constants.AGG_MONITORING_RESOURCE, conv.Audit.StreamingStats.AggMonitoringResources.DashboardName, string(aggMonitoringResourcesBytes))
+		if err != nil {
+			return err
+		}
+		err = txn.BufferWrite([]*spanner.Mutation{mutation})
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		err = fmt.Errorf("can't store aggregate monitoring resources for migration job %s: %v", migrationJobId, err)
+		return err
+	}
+	return nil
+}
+
+// PersistResources stores all the metadata associated with a shard orchestration for a minimal downtime migration in the metadata db. An example of this metadata is generated resources.
+func PersistResources(ctx context.Context, targetProfile profiles.TargetProfile, sourceProfile profiles.SourceProfile, conv *internal.Conv, migrationJobId string, dataShardId string) (err error) {
```

Comment:
Why are there different persistResources and writeJobResources method ? It sounds very similar

Reply:
`PersistResources` is an externally visible API interface for this package, `writeJobResources` is an internal, package-local method.